app.go

```diff
@@ -2,20 +2,24 @@ package passage
 
 import (
 	"context"
+	"fmt"
 
-	jwkLibrary "github.com/lestrrat-go/jwx/jwk"
+	"github.com/lestrrat-go/jwx/v2/jwk"
 )
 
+const jwksUrl = "https://auth.passage.id/v1/apps/%v/.well-known/jwks.json"
+
 type Config struct {
 	APIKey     string
 	HeaderAuth bool
 }
 
 type App struct {
-	ID     string
-	JWKS   jwkLibrary.Set
-	Config *Config
-	client *ClientWithResponses
+	ID        string
+	JWKS      jwk.Set
+	Config    *Config
+	client    *ClientWithResponses
+	jwksCache *jwk.Cache
```

Comment:
It looks like this is using 2 data structures (`jwk.Set` and `jwk.Cache`) to store jwk info.

Would it be possible to use [jwk.CachedSet](https://github.com/lestrrat-go/jwx/blob/8d1d78351e9f259723f9f558889a78e327379683/examples/jwk_cached_set_example_test.go#L27) & store it in 1?

Reply:
I tried it at first. It's a slim immutable shim over Set. But the blocker was that it didn't allow access to the underlying cache, which is needed to perform the refresh.

Reply:
It looks like the structure abstracts away the need to refresh manually (in the doc [here](https://github.com/lestrrat-go/jwx/blob/develop/v3/docs/04-jwk.md#auto-refreshing-remote-keys)), though that'd require some other refactoring.

& it's not necessary to solve the thread-safe issue, so we can keep this change focused on the fix.

Reply:
it was originally required to fulfill the re-fetch logic https://github.com/passageidentity/passage-go/blob/8cb184a60bcced14821b84fad7ae2507a4abe85b/authentication.go#L45-L55

but given our recent discussions around dropping it, it can be refactored to use the `CachedSet` instead. though i wanted to tackle removing the re-fetch logic in a separate PR since it's a functional change.